xformers/factory/block_configs.py

```diff
@@ -0,0 +1,224 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
```

Comment:
99% of the code is the same as  before, just moved out of "block_factory" which was becoming a little big. A lot of the steps here are pure config machinery, so I figured that block_factory was easier to understand without them

Reply:
Nice, feels cleaner!